src/imitation/scripts/train_dagger.py

```diff
@@ -0,0 +1,160 @@
+        total_timesteps: The number of timesteps to train inside the environment. (In
+            practice this is a lower bound, as the number of timesteps is rounded up
+            to finish a DAgger training rounds.)
+        batch_size: Number of observation-action samples used in each BC update.
+        bc_train_kwargs: The `bc_train_kwargs` argument for
+            `SimpleDAggerTrainer.train()`. A dict of keyword arguments that are passed
+            to `BC.train()` every DAgger training round.
+        l2_weight: L2 regularization weight for BC.
+        optimizer_cls: The Torch optimizer class used for BC updates.
+        optimizer_kwargs: Optimizer kwargs passed to BC.
+        log_dir: Log output directory. Final policy is also saved in this directory as
+            "{log_dir}/final.pkl"
+        n_episodes_eval: The number of final evaluation rollout episodes, if `venv` is
+            provided. These rollouts are used to generate final statistics saved into
+            Sacred results, which can be compiled into a table by
+            `imitation.scripts.analyze.analyze_imitation`.
+    """
+    # TODO(shwang): Add support for directly loading a BasePolicy `*.th` file.
+    log_dir = pathlib.Path(log_dir)
+    log_dir.mkdir(parents=True, exist_ok=True)
+    logging.info("Logging to %s", log_dir)
+
+    logger.configure(log_dir, ["tensorboard", "stdout"])
+    sacred_util.build_sacred_symlink(log_dir, _run)
+
+    if expert_policy_path is None:
+        raise ValueError("expert_policy_path cannot be None")
+
+    expert_policy = serialize.load_policy(expert_policy_type, expert_policy_path, venv)
+    if not isinstance(expert_policy, policies.BasePolicy):
+        raise TypeError(f"Unexpected type for expert_policy: {type(expert_policy)}")
+
+    expert_trajs: Sequence[types.Trajectory]
+    if expert_data_src_format == "path":
+        expert_trajs = types.load(expert_data_src)
+    elif expert_data_src_format == "trajectory":
+        # Convenience option for launching experiment from Python script with
+        # in-memory trajectories.
+        expert_trajs = expert_data_src
+    else:
+        raise ValueError(
+            f"expert_data_src_format={expert_data_src_format} should be 'path'"
+            "or 'trajectory'."
+        )
+
+    for x in expert_trajs:
+        if not isinstance(x, types.Trajectory):
+            raise ValueError(f"Expert data is the wrong type: {type(x)}")
+
+    # TODO(shwang): Copied from scripts/train_adversarial -- refactor with "auto"?
+    if n_expert_demos is not None:
+        if not len(expert_trajs) >= n_expert_demos:
+            raise ValueError(
+                f"Want to use n_expert_demos={n_expert_demos} trajectories, but only "
+                f"{len(expert_trajs)} are available."
+            )
+        expert_trajs = expert_trajs[:n_expert_demos]
+
+    # pytype is getting super confused about the type of expert_trajs and doesn't
```

Comment:
The problem is you made `DAggerTrainer` expect a `List` (when in fact any `Sequence` would do), so it actually didn't type check as you declared `expert_trajs` here to be a `Sequence`. I've fixed this now and removed the `# pytype: disable`. I'd in general suggest using the most generic type in parameter and return annotations to keep the API as assumption-free as possible.

Reply:
Oh nice, thanks for catching that!